Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.App.KronosWfc.BusinessLogic/Common/XmlHelper.cs

```diff
@@ -37,5 +39,90 @@ public static T ProcessResponse<T>(this Tuple<string, string> response, Telemetr
             var xResponse = xDoc.Root.Descendants().FirstOrDefault(d => d.Name.LocalName.Equals(Response, StringComparison.Ordinal));
             return xResponse.ToString().DeserializeObject<T>();
         }
+
+        /// <summary>
+        /// Create Kronos comments ensuring we preserve previous notes.
+        /// </summary>
+        /// <param name="noteMessage">The note to add.</param>
+        /// <param name="noteCommentText">The comment text value of the note to add.</param>
+        /// <param name="existingNotes">Existing notes.</param>
+        /// <returns>Kronos Comments object.</returns>
+        public static Comments GenerateKronosComments(string noteMessage, string noteCommentText, List<Comment> existingNotes = null)
+        {
+            var comments = new Comments
+            {
+                Comment = new List<Comment>(),
+            };
+
+            if (existingNotes != null)
+            {
+                comments.Comment.AddRange(existingNotes);
+            }
+
+            comments.Comment.Add(new Comment
+            {
+                CommentText = noteCommentText,
+                Notes = new List<Notes>
+                {
+                    new Notes
+                    {
+                        Note = new List<Note>
+                        {
+                            new Note
+                            {
+                                Text = noteMessage,
+                                Timestamp = DateTime.UtcNow.ToString(CultureInfo.InvariantCulture),

```

Comment:
I chose UTC because the way we handle comments makes these timestamps pretty pointless. I agree though it would be misleading having comments originating from teams in UTC and comments from Kronos in local so I have made the change.